src/church/lists.rs

```diff
@@ -474,6 +474,43 @@ pub fn filter() -> Term {
     )
 }
 
+/// Applied to a Church-encoded list it returns the last element.
+///
+/// LAST := Z (λzl.NULL l (λx.NIL) (λx.NULL (tail l) (head l) (z (tail l))) I) =
+/// Z (λ 2 1. NULL 1 (λ NIL) (λ NULL (tail 2) (head 2) (3 (tail 2))) I)
+///
+/// # Example
+/// ```
+/// # #[macro_use] extern crate lambda_calculus;
+/// # fn main() {
+/// use lambda_calculus::church::lists::{last};
+/// use lambda_calculus::*;
+///
+/// let list = Term::from(vec![0.into(), 1.into(), 2.into(), 3.into()]);
+///
+/// assert_eq!(beta(app!(last(), list.clone()), NOR, 0, false), 3.into());
```

Comment:
The `clone()` is unnecessary here; it is added when more than a single doctest is performed on the same `Term`, but it's ok to consume the `list` here.

Reply:
You can use `app()` instead of `app!()` here - that way you can simplify the doctest's boilerplate; That reminds me - I'll do the same with the other doctests.

Reply:
I just did and actually it looks like now you can remove the whole 
    /// # #[macro_use] extern crate lambda_calculus;
    /// # fn main() {
    ...
    /// # }
wrappers regardless of whether you use macros or not.